source/extensions/upstreams/http/http/upstream_request.h

```diff
@@ -26,6 +27,9 @@ class HttpConnPool : public Router::GenericConnPool, public Envoy::Http::Connect
     conn_pool_ = cm.httpConnPoolForCluster(route_entry.clusterName(), route_entry.priority(),
                                            downstream_protocol, ctx);
   }
+  ~HttpConnPool() override {
+    RELEASE_ASSERT(conn_pool_stream_handle_ == nullptr, "conn_pool_stream_handle not null");
```

Comment:
Preference for ASSERT, or you can use ENVOY_BUG if you think it's tracking-worthy.

Reply:
Fair, updated.